server/requeststojointeamhandler_test.go

```diff
@@ -0,0 +1,176 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/fluidkeys/api/datastore"
+	"github.com/fluidkeys/fluidkeys/assert"
+	"github.com/fluidkeys/fluidkeys/exampledata"
+	"github.com/fluidkeys/fluidkeys/fingerprint"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+	"github.com/gofrs/uuid"
+)
+
+func TestListRequestsToJoinTeamHandler(t *testing.T) {
+
+	teamUUID, err := uuid.FromString("74bb40b4-3510-11e9-968e-53c38df634be")
+	assert.NoError(t, err)
+
+	goodRoster := `
+uuid = "74bb40b4-3510-11e9-968e-53c38df634be"
+
+[[person]]
+email = "test4@example.com"
+fingerprint = "BB3C 44BF 188D 56E6 35F4  A092 F73D 2F05 33D7 F9D6"
+`
+	unlockedKey, err := pgpkey.LoadFromArmoredEncryptedPrivateKey(
+		exampledata.ExamplePrivateKey4, "test4")
+
+	now := time.Date(2019, 2, 10, 16, 35, 45, 0, time.UTC)
+
+	goodSignature, err := unlockedKey.MakeArmoredDetachedSignature([]byte(goodRoster))
+	assert.NoError(t, err)
+
+	goodTeam := datastore.Team{
+		UUID:            teamUUID,
+		Roster:          goodRoster,
+		RosterSignature: goodSignature,
+		CreatedAt:       now,
+	}
+
+	setup := func() {
+		assert.NoError(t,
+			datastore.UpsertPublicKey(nil, exampledata.ExamplePublicKey4))
+
+		assert.NoError(t,
+			datastore.LinkEmailToFingerprint(
+				nil, "test4@example.com", exampledata.ExampleFingerprint4,
+			),
+		)
+
+		assert.NoError(t,
+			datastore.CreateTeam(nil, goodTeam),
+		)
+	}
+
+	teardown := func() {
+		_, err := datastore.DeletePublicKey(exampledata.ExampleFingerprint4)
+		assert.NoError(t, err)
+
+		_, err = datastore.DeleteTeam(nil, teamUUID)
+		assert.NoError(t, err)
+	}
+
+	setup()
+	defer teardown()
+
+	t.Run("lists requests from valid fingerprint", func(t *testing.T) {
+		requestToJoinUUID, err := datastore.CreateRequestToJoinTeam(
+			nil,
+			teamUUID,
+			"request@example.com",
+			fingerprint.MustParse("AAAABBBBAAAABBBBAAAABBBBAAAABBBBAAAABBBB"),
+			now,
+		)
+		assert.NoError(t, err)
+
+		response := callAPI(
+			t,
+			"GET",
+			fmt.Sprintf("/v1/team/%s/requests-to-join", teamUUID),
+			&exampledata.ExampleFingerprint4,
+		)
+
+		t.Run("status code 200", func(t *testing.T) {
+			assertStatusCode(t, http.StatusOK, response.Code)
+		})
+
+		t.Run("response has JSON content type", func(t *testing.T) {
+			// TODO: check server returned content-type: application/json (and elsewhere!)
+		})
+
+		t.Run("response body has requests JSON", func(t *testing.T) {
+			expected := `{
+    "requests": [
+        {
+            "uuid": "` + requestToJoinUUID.String() + `",
+            "fingerprint": "AAAA BBBB AAAA BBBB AAAA  BBBB AAAA BBBB AAAA BBBB",
```

Comment:
ah here's a trap, because we haven't been consistent in our roster (but we should fix that):

to be future proof we should use `Fingerprint.Uri()` which is the format `OPENFPR4:AAAABBBBAAAABBBBAAAABBBBAAAABBBBAAAABBBB`

it should parse and format just the same

Reply:
Ah hah - thanks!